src/openforms/registrations/contrib/json_dump/plugin.py

```diff
@@ -0,0 +1,142 @@
+import base64
+
+from django.utils.translation import gettext_lazy as _
+
+from zgw_consumers.client import build_client
+
+from openforms.formio.typing import Component
+from openforms.forms.utils import to_json
+from openforms.submissions.models import (
+    Submission,
+    SubmissionFileAttachment,
+    SubmissionValueVariable,
+)
+from openforms.typing import JSONObject
+from openforms.variables.constants import FormVariableSources
+from openforms.variables.validators import validate_path_context_values
+
+from ...base import BasePlugin  # openforms.registrations.base
+from ...registry import register  # openforms.registrations.registry
+from .config import JSONDumpOptions, JSONDumpOptionsSerializer
+
+
+@register("json_dump")
+class JSONDumpRegistration(BasePlugin):
+    verbose_name = _("JSON dump registration")
+    configuration_options = JSONDumpOptionsSerializer
+
+    def register_submission(
+        self, submission: Submission, options: JSONDumpOptions
+    ) -> dict:
```

Comment:
bare dicts are not very useful because you have no idea about the key type or the value type, so at the minimum I would declare this as `dict[str, Any]`. However, we know that the return value must be JSON serializable because it gets stored in a `JSONField`, so you can use `openforms.typing.JSONObject` as type hint which limits the keys to `str` type and the values to primitives & json objects/arrays.

```suggestion
) -> JSONObject:
```

Reply:
for future checking: type checker didn't agree :)